docker-compose.yml

```diff
@@ -1,35 +1,120 @@
+    DOCSRS_BUILD_CPU_LIMIT: 2
+    DOCSRS_INCLUDE_DEFAULT_TARGETS: false
+
+x-builder: &builder
+    build:
+        context: .
+        dockerfile: ./dockerfiles/Dockerfile
+        target: build-server
+    depends_on:
+        - db
+        - s3
+    environment: *environment
+    env_file:
+        - .docker.env
+    healthcheck:
+        << : *healthcheck-interval
+        test: curl --silent --fail localhost:3000/about/metrics
+
 services:
     web:
         build:
             context: .
             dockerfile: ./dockerfiles/Dockerfile
+            target: web-server
         platform: "linux/amd64"
         depends_on:
             - db
             - s3
         ports:
-            - "3000:3000"
-        # for metrics
-        expose: ["3000"]
+            - "3000:80"
+        environment: *environment
+        env_file:
+            - .docker.env
+        healthcheck:
+            << : *healthcheck-interval
+            test: curl --silent --fail localhost:80/about/metrics
+
+    # Include the registry watcher with `docker compose --profile watch up -d`
+    registry-watcher:
+        build:
+            context: .
+            dockerfile: ./dockerfiles/Dockerfile
+            target: registry-watcher
+        platform: "linux/amd64"
+        depends_on:
+            - db
         volumes:
-            - "/var/run/docker.sock:/var/run/docker.sock"
-            - ".rustwide-docker:/opt/docsrs/rustwide"
             - "cratesio-index:/opt/docsrs/prefix/crates.io-index"
-        environment:
-            DOCSRS_RUSTWIDE_WORKSPACE: /opt/docsrs/rustwide
-            DOCSRS_DATABASE_URL: postgresql://cratesfyi:password@db
-            DOCSRS_STORAGE_BACKEND: s3
-            S3_ENDPOINT: http://s3:9000
-            AWS_ACCESS_KEY_ID: cratesfyi
-            AWS_SECRET_ACCESS_KEY: secret_key
+        environment: *environment
         env_file:
-            - .env
+            - .docker.env
+        profiles:
+            - watch
+            - all
         healthcheck:
-            test: ["CMD", "curl", "--silent", "--fail", "localhost:3000"]
-            interval: 10s
-            timeout: 5s
-            retries: 10
+            << : *healthcheck-interval
+            test: curl --silent --fail localhost:3000/about/metrics
+
+    builder-a:
+        << : *builder
+        volumes:
+            - ".rustwide-docker/builder-a:/opt/docsrs/rustwide"
+            - "/var/run/docker.sock:/var/run/docker.sock"
+
+    builder-b:
+        << : *builder
+        volumes:
+            - ".rustwide-docker/builder-b:/opt/docsrs/rustwide"
+            - "/var/run/docker.sock:/var/run/docker.sock"
+
+    cli:
+        build:
+            context: .
+            dockerfile: ./dockerfiles/Dockerfile
+            target: cli
+        depends_on:
+            - db
+            - s3
+        volumes:
+            - ".rustwide-docker/cli:/opt/docsrs/rustwide"
+            - "cratesio-index:/opt/docsrs/prefix/crates.io-index"
```

Comment:
That was how I had it previously, till updating the readme to use `cli` instead of `web` for running `cratesfyi build crate`. I guess I could change that to use a build stage image instead though.

Reply:
Updated that, with the `ENTRYPOINT` changes it's now possible to easily use the build-image to run builds (and the `registry-watcher` image to run commands that need the `crates.io-index`).